### DISCUSSION TITLE
Feature Request: pager support

### DISCUSSION THREAD

**Discussion by jonascarpay** [duckdb/duckdb#13427] [Ideas]
I was surprised to see duckdb doesn't have any support for pagers, [pspg](https://github.com/okbob/pspg) in particular. Is there a reason for this? Currently I work around this by piping CSV into it, but it would be nice if it were able to integrate into the duckdb repl in the same way that psql does.

**Comment by uosis** [comment]
I was also surprised by the lack of pager. Seems like a small effort for a big usability improvement. Piping to external command is not really practical during data exploration which involves running multiple adhoc queries.

**Comment by aszenz** [comment]
I would appreciate pager support in duckdb cli as well, i normally use mycli which supports piping to an external pager, see:

https://www.mycli.net/pager

Without pager support it's difficult to explore tables with lot of columns

**Comment by tobwen** [comment]
Ha! Just had the same problem.

**Comment by tobwen** [comment]
Good news, I just developed a prototyp for paging! It works already. Need to fix some stuff, I'll PR it then: https://github.com/duckdb/duckdb/pull/19004

**Reply by aszenz** [reply]
Looking forward to it

**Reply by tobwen** [reply]
There it is: https://github.com/duckdb/duckdb/pull/19004

**Comment by tobwen** [comment]
I think I'll remove `auto` and use the environment variable only by default. That'll stop future discussions :D

In my private branch, I've already some heuristics to make it work even better.

**Comment by tobwen** [comment]
Massive update on this one! https://github.com/duckdb/duckdb/pull/19004#issuecomment-3368007025

**Reply by EduardoVernier** [reply]
that's amazing, can't wait to use it with visidata
